app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt

```diff
@@ -87,36 +89,49 @@ object DraftController {
         }?.localUuid
     }
 
-    fun MutableRealm.setDraftSignature(draftLocalUuid: String) {
-        updateDraft(draftLocalUuid, this) { draft ->
+    fun MutableRealm.setSignature(draft: Draft) {
 
-            draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
+        draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
 
-            val defaultSignature = SignatureController.getDefaultSignature(this) ?: return@updateDraft
+        SignatureController.getDefaultSignature(this)?.let { defaultSignature ->
 
-            if (draft.identityId == null) draft.identityId = defaultSignature.id
+            draft.identityId = defaultSignature.id
 
-            if (draft.from.isEmpty()) {
-                draft.from = realmListOf(Recipient().apply {
-                    this.email = defaultSignature.sender
-                    this.name = defaultSignature.fullName
-                })
+            draft.from = realmListOf(Recipient().apply {
+                this.email = defaultSignature.sender
+                this.name = defaultSignature.fullName
+            })
+
+            draft.replyTo = realmListOf(Recipient().apply {
+                this.email = defaultSignature.replyTo
+                this.name = ""
+            })
+
+            val html = "<br/><br/><div class=\"editorUserSignature\">${defaultSignature.content}</div>"
+            draft.body = when (defaultSignature.position) {
+                SignaturePosition.AFTER_REPLY_MESSAGE -> draft.body + html
+                else -> html + draft.body
             }
+        }
```

Comment:
I would see it rather in a method init of Draft, at its creation we give it a signature

Reply:
This function has been moved to the `Draft` object.